ru/lessons/advanced/gen-stage.md

```diff
@@ -0,0 +1,261 @@
+---
+layout: page
+title: GenStage
+category: advanced
+order: 11
+lang: ru
+---
+
+В этом уроке мы поближе рассмотрим GenStage, узнаем, какую роль он выполняет, и как мы можем использовать его в наших приложениях.
+
+{% include toc.html %}
+
+## Вступление
+
+Так что же такое GenStage? В официальной документации написано, что это "спецификация и поток вычислений для Elixir", но что это означает для нас?
+
+Это означает, что GenStage предоставляет нам возможность определить конвейер задач, который будет выполняться независимыми шагами (или этапами) в отдельных процессах. Если вы уже работали с конвейерами, тогда некоторые из этих понятий могут быть вам знакомы.
+
+Чтобы лучше понять, как это работает, давайте визуализируем простой поток производитель-потребитель:
+
+```
+[A] -> [B] -> [C]
+```
+
+В этом примере у нас три стадии: `A` — производитель, `B` — производитель-потребитель, и `C` — потребитель.  `A` производит значение, которое потребляет `B`, `B` выполняет какую-то работу и возвращает значение, которое отправляется нашему потребителю `C`. Роль нашей стадии важна, что мы увидим в следующей секции.
+
+Пока в нашем примере производитель относится к потребителю один к одному, но можно одновременно иметь несколько производителей и множество потребителей на любой стадии.
+
+Чтобы лучше проиллюстрировать эти понятия, мы построим конвейер с помощью GenStage, но сначала давайте рассмотрим чуть подробнее роли, на которые GenStage опирается.
+
+## Производитель и потребитель
+
+Как мы уже рассмотрели, роль которую мы даем нашей стадии важна. Спецификация GenStage принимает три роли:
+
++ `:producer` — Источник.  Производители ждут требования от потребителей и реагируют, выполняя запрошенные действия.
+
++ `:producer_consumer` — И источник, и поглотитель.  Производитель-потребитель может как реагировать на требования от потребителей, так и запрашивать выполнение действий от производителей.
+
++ `:consumer` — Поглотитель. Потребитель запрашивает и получает данные от производителя.
+
+Заметили, что наши производители _ожидают_ требования?  Благодаря GenStage наши потребители посылают требование вверх по потоку и обрабатывают данные от производителя.  Это облегчает механизм, который называется обратное давление.  Обратное давление перекладывает ответственность на неперегруженного производителя, когда потребители заняты.
+
+Теперь, когда мы рассмотрели роли в GenStage, займёмся нашим приложением.
+
+## Начало работы
+
+В этом примере мы соберем GenStage приложение, которое генерирует числа, выбирает четные и в конце печатает их.
```

Comment:
Yup. Just replace the stuff, don't write it in comments. :)

Replacing would be faster way to go.